server/src/main/java/org/opensearch/index/engine/EngineConfig.java

```diff
@@ -133,9 +134,13 @@ public Supplier<RetentionLeases> retentionLeasesSupplier() {
             case "lz4":
             case "best_compression":
             case "zlib":
+            case "lucene_default":
+                return s;
             case "zstd":
             case "zstd_no_dict":
-            case "lucene_default":
+                if (!FeatureFlags.isEnabled(FeatureFlags.ZSTD_COMPRESSION)) {
```

Comment:
nit...explicit logic check, negations are hard to track and often overlooked (we should add this as a rule in spotless). 
```suggestion
                if (FeatureFlags.isEnabled(FeatureFlags.ZSTD_COMPRESSION) == false) {
```